ofrak_core/ofrak/ofrak_context.py

```diff
@@ -110,9 +110,19 @@ async def create_ofrak_context(self) -> OFRAKContext:
         component_locator = await self.injector.get_instance(ComponentLocatorInterface)
 
         resource_factory = await self.injector.get_instance(ResourceFactory)
-        component_locator.add_components(
-            await self.injector.get_instance(List[ComponentInterface]), self._discovered_modules
+        component_auditor = _ComponentDepAuditor()
+        components = [
```

Comment:
I think components should only be audited when a specific flag is passed. Perhaps add a note in the missing component error message, suggesting this option. By default I believe all components should be imported and used.

Reply:
To clarify: if all components are imported and used as you suggest, then there will be a dependency error when the user tries to actually run the component, right? Whereas if they're audited at the beginning, some dependency injector stuff will fail earlier in the script.

As an OFRAK user, I'd probably rather know about any errors in my script as soon as possible since complicated scripts can take a while. If I'm understanding correctly, that could be a case for auditing as early as possible.

Reply:
I agree with jacob here, especially during development, on occasion waiting 5+ minutes to see that a typo made it into my docker and failed the import stage, not hitting the error until component lookup.

Reply:
So during development, use that flag to audit components. I'm arguing that the default behavior should simply import and use the components as the user directed it to (with `.discover`), and auditing should be an option that needs to be toggled on. I think our default assumption should be that OFRAK is installed correctly, and code to deal with cases where it is not (especially code that increases runtime) should not be run by default.

I'm not sure how dependency auditing will help with typos in components though, so maybe I'm misunderstanding the conversation here.